packages/adapter/lib/adapter/adapter.js

```diff
@@ -8091,9 +8093,8 @@ function Adapter(options) {
                     .map(id => (typeof id === 'string' && !id.startsWith(ALIAS_STARTS_WITH) ? id : null))
                     .filter(id => id);
 
-                const promises = [];
-
                 for (let aliasPattern of pattern) {
+                    // TODO according to unsubscribe and jsdoc regexp is not allowed as array - we should return here with an error
```

Comment:
According to JSDOC we allow `string`, `string[]`, `RegExp`, also `unsubscribeForeignStates` does not support Array of Regexps code-wise, we should throw/error cb here. Other opinions? Also RegExp in general is weird if we just stringify it and then use it as pattern, because db code and Redis won't understand RegExp correctly

Reply:
Then yes we should fix that. Hopefully we do not hit too many adapters

Reply:
Or maybe better then throwing adding a deprecation log and "ignore it" (because also former ist was kind of not effective"?